server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1213,6 +1215,11 @@ public Cluster updateCluster(UpdateClusterCmd cmd) {
             }
         }
 
+        if (architecture != null) {
```

Comment:
meaning it can never be unset?

Reply:
It cannot be unset but can be changed. Please note from the upgrade schema, every existing cluster pre-upgrade will have a set default value (there will be no unset arch values)